sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2009,12 +2020,200 @@ __ESIMD_API void media_block_store(AccessorTy acc, unsigned x, unsigned y,
 }
 #endif // !__ESIMD_FORCE_STATELESS_MEM
 
+/// Variant of block_load that uses local accessor as a parameter
+/// Loads a contiguous block of memory from given accessor and offset and
+/// returns the loaded data as a vector. Actual code generated depends on
+/// the alignment parameter.
+/// @tparam Tx Element type.
+/// @tparam N Number of elements to load, <code>N * sizeof(Tx)</code> must
+/// be
+///    1, 2, 4 or 8 owords long.
+/// @tparam AccessorTy Accessor type (auto-deduced).
+/// @tparam Flags The alignment specifier type tag. Auto-deduced from the
+///    \c Flags parameter. If it is less than \c 16, then slower unaligned
+///    access is generated, othewise the access is aligned.
+/// @param acc The accessor.
+/// @param offset The offset to load from in bytes.
+/// @return A vector of loaded elements.
+///
+template <typename Tx, int N, typename AccessorTy,
```

Comment:
Do we need the same restrictions as in the normal accessor version, like `(N == 8 || N == 16 || N == 32) && sizeof(T) == 4 `

Reply:
it looks like this current function doesn't have any checks, as long as it's checked somewhere that's fine with me, but IMO we should be consistent for these new functions, either do all the checks here or leave all of them to the slm_ functions